app/controllers/posts_controller.rb

```diff
@@ -3,11 +3,43 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @page = params.fetch(:page, 0).to_i
-    @posts = Post.limit(3).offset(@page * 3)
+    @posts = @user.posts.limit(3).offset(@page * 3)
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build
   end
 
   def show
     @post = Post.find(params[:id])
     @user = @post.author
+    @likes_count = @post.likes_counter
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.create(user: current_user, post: @post)
+
+    respond_to do |format|
+      format.json { render json: { likes_count: @post.likes.count } }
+    end
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
```

Comment:
[OPTIONAL]
- [ ] According to the project requirement we should created a post on behalf of the currently logged in user. Therefore I suggest that you make use of the `current_user` method that you created in the `applicationController` to set the user for a new post.
  
Example:
```rb
@post = current_user.posts.build(post_params)
```

_This is a requirement of the project but first reviewer didn't notice it. so I requested this as an optional but I highly recommend you to make the change_